### PR TITLE
Feature: Multiple routing profiles

### PR DESCRIPTION
**Pull request by Fsss126**
This PR possibility to build route composed from segments with different transportation profiles via passing array of profiles as second argument to `setWaypoints`.

- Currently interactive mode is not supported when using composed routing

### CODE REVIEW

File: demo/src/examples/14 Multiple profiles.svelte [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    directions?.setWaypoints(waypoints, shuffle(profiles));
+  }
+</script>
+
+<AppSidebar>
+  <span slot="title">{meta.name}</span>
+  <p>
+    This example showcases routing with multiple profiles. Segments corresponding to different profiles are displayed in
+    different colors. Plugin provides default styles for typical <a
+      href="https://project-osrm.org/docs/v5.5.1/api/#general-options">OSRM profiles</a
```
Comment (smellyshovel) [inline demo/src/examples/14 Multiple profiles.svelte:64]: Add a `target=_blank` attribute so that the "chain" emoji is automatically added and the link opens in a new tab (for an example see the "Source" links on the top of demo pages).

File: demo/src/examples/14 Multiple profiles.svelte [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    >: car, bike, foot. Styles can be changed per profile via general style customization approach (consult the
+    <a href="/examples/restyling" use:link>Restyling example</a>). In case different profiles are used you can similarly
+    style map features corresponding to each profile by targeting profile property of a feature (see
+    <a href="https://github.com/maplibre/maplibre-gl-directions/blob/main/src/directions/layers.ts">default styles</a>).
```
Comment (smellyshovel) [inline demo/src/examples/14 Multiple profiles.svelte:68]: The same as above for this link as well, please.

File: demo/src/examples/14 Multiple profiles.svelte
Comment (smellyshovel) [inline demo/src/examples/14 Multiple profiles.svelte:1]: Would be also really nice, if you specified somewhere in the example's description text which colors of the routing line represent which profiles on this Demo page.

File: package.json [hunk 3]
```diff
@@ -35,7 +35,7 @@
   "scripts": {
     "prepare": "husky install",
     "env:prep": "npm run build:lib && npm link && npm link @maplibre/maplibre-gl-directions",
-    "dev:lib": "npm run check:lib && tsc --project ./tsconfig.lib.json && vite build --watch --mode development --config vite.lib.config.ts",
```
Comment (smellyshovel) [inline package.json:38]: Why did you remove the `npm run check:lib` part?

File: package.json [hunk 4]
```diff
@@ -56,8 +58,7 @@
   },
   "dependencies": {
     "@placemarkio/polyline": "^1.2.0",
-    "nanoid": "^4.0.2",
-    "svelte-spa-router": "^3.3.0"
```
Comment (smellyshovel) [inline package.json:60]: Why moving the router to dev dependencies?

File: src/directions/main.ts [hunk 5]
```diff
@@ -919,21 +971,34 @@ export default class MapLibreGlDirections extends MapLibreGlDirectionsEvented {
...
-        this.configuration.bearings
-          ? {
-              bearing: this.waypointsBearings[index],
-            }
-          : undefined,
-      );
+    this.profiles = profiles.slice(0, waypoints.length - 1);
+    if (this.profiles.length === 0) {
+      this.profiles.push(this.configuration.profile);
+    }
```
Comment (smellyshovel) [inline src/directions/main.ts:983]: Please, add comments here for the future readers, explaining the logic of what that code's doing.

File: src/directions/main.ts [hunk 6]
```diff
@@ -99,6 +107,26 @@ export default class MapLibreGlDirections extends MapLibreGlDirectionsEvented {
     });
   }
 
+  protected async fetch(data: RequestData) {
+    const { method, url, payload } = data;
```
Comment (smellyshovel) [inline src/directions/main.ts:111]: Can be shortened to just

```
  protected async fetch({ method, url, payload }: RequestData) {
```

File: src/directions/helpers.ts [hunk 7]
```diff
@@ -62,3 +63,17 @@ export function coordinatesComparator(
...
+  });
+}
+
+export function getWaypointsBearings(waypoints: Feature<Point>[]): ([number, number] | undefined)[] {
+  return waypoints.map((waypoint) => {
+    return Array.isArray(waypoint.properties?.bearing)
+      ? [waypoint.properties?.bearing[0], waypoint.properties?.bearing[1]]
+      : undefined;
+  });
+}
```
Comment (smellyshovel) [inline src/directions/helpers.ts:79]: Please, add the doc-comments for the added methods. Look above for inspiration and examples

**Review by smellyshovel — CHANGES_REQUESTED** [review]
Overall everything's fine. Looking forward for you to resolve the minor points I marked. Thanks for the awesome contribution! :+1:

File: src/directions/main.ts (hunk 5 above)
Comment (smellyshovel) [reply src/directions/main.ts:983]: Actualy, would also be nice to add some comments to the `fetch`/`fetchDirections` parts as well, but, I guess, that's optional, since (if I'm not mistaken) there were none before your contribution. But I'd appreciate them anyway, or otherwise there's a risk that future readers won't understand a thing that's going on there (it's already kinda hard to get the grasp of).

File: package.json (hunk 4 above)
Comment (Fsss126) [reply package.json:60]: Because it's used only in demo app and isn't needed for the shipped lib

File: package.json (hunk 3 above)
Comment (Fsss126) [reply package.json:38]: Running check once before start of dev server has little sense cause check is not watching for changes. I added tsc:lib:watch instead since it's needed to regenerate typings on code changes for correct types to be discovered in the demo app during development. I simply ran it in a separate terminal tab but it might be called from the dev:lib with run-p to start it in parallel with dev server if you deem necessary.

File: package.json (hunk 4 above)
Comment (smellyshovel) [reply package.json:60]: But the demo app gets deployed as well, as a separate application. Isn't that important?

File: package.json (hunk 3 above)
Comment (smellyshovel) [reply package.json:38]: Obviously it gets run only once, when starting the dev server. As far as I know, there's no way to run in continuosly (sort of a watch-mode). But running it once is anyway better than not running it a all. I think this check shouldn't be removed.

File: demo/src/examples/14 Multiple profiles.svelte (hunk 1 above)
Comment (Fsss126) [reply demo/src/examples/14 Multiple profiles.svelte]: Done

File: demo/src/examples/14 Multiple profiles.svelte (hunk 2 above)
Comment (Fsss126) [reply demo/src/examples/14 Multiple profiles.svelte]: Done

File: demo/src/examples/14 Multiple profiles.svelte
Comment (Fsss126) [reply demo/src/examples/14 Multiple profiles.svelte:1]: Improved the demo to display currently applied profiles with their color.

File: package.json (hunk 3 above)
Comment (Fsss126) [reply package.json:38]: But what's the point in checking before start? It's called during build, this should be enough.

Comment (Fsss126) [reply package.json:38]: BTW watch option has been added to `svelte-check` https://github.com/sveltejs/language-tools/issues/287

File: package.json (hunk 4 above)
Comment (Fsss126) [reply package.json:60]: No, it's deployed as a static build so this doesn't matter.

File: src/directions/main.ts (hunk 5 above)
Comment (Fsss126) [reply src/directions/main.ts]: Described some crucial logic in the comments

File: src/directions/main.ts (hunk 6 above)
Comment (Fsss126) [reply src/directions/main.ts]: Changed it

File: src/directions/helpers.ts (hunk 7 above)
Comment (Fsss126) [reply src/directions/helpers.ts]: Added

File: package.json (hunk 3 above)
Comment (smellyshovel) [reply package.json:38]: So, taking the newly-added `watch` mode into account, I have two options for you:
1. Either configure the script properly to run both the server and the checker alongside each other (in parallel?)
2. Or restore the removed one-time check

I personally use it quite often with just restarting the dev server. At least because it's way faster than building the whole package just to check the Svelte correctness.

That's the last requirement before we're good to go with this PR.

**Review by smellyshovel — APPROVED** [review]
